Refactor: use errors.Join for multi-error returns

hakkiai opened the pull request:
What this PR does 

This PR updates places in the codebase where multiple errors were being combined using fmt.Errorf("...: %v", errs).
They now use errors.Join(errs...) instead.

errors.Join keeps the original error types intact and allows unwrapping, while the previous formatting approach was losing that context.

**Issue fixed**

- Fixes #191

**Notes for the reviewer**

Files/areas touched in this refactor:

1. Controller: bucketaccess.go

2. Sidecar: bucket.go

3. Internal protocol: s3.go, azure.go, gcs.go

**Verification**

Checked that the Validate methods now return joined errors instead of formatted slices.

All existing tests passed after the change.

BlaineEXE commented on sidecar/internal/reconciler/bucket.go:
```diff
@@ -342,37 +343,37 @@ func objectProtocolListFromApiList(apiList []cosiapi.ObjectProtocol) ([]*cosipro
 
 	if len(unsupportedProtos) > 0 {
-		return fmt.Errorf("driver %q does not support protocols: %v", driver.name, unsupportedProtos)
+		return fmt.Errorf("driver %q does not support protocols: %w", driver.name, errors.Join(unsupportedProtos...))
```
This is just a list of unsupported protocols, not necessarily errors. Let's revert this small set of changes.

BlaineEXE reviewed (changes requested):
Thanks for the contribution. There are a couple minor functions I'd like to leave unchanged, but everything else looks great.

Addendum: please ensure there is a single (squashed) commit after followup changes.

hakkiai commented:
Thanks for the review @BlaineEXE 

I've reverted the `unsupported` logic back to using `[]string` as requested. because this matches the original code, those specific sections no longer appear in the diff.

I've kept `errors.Join` for the actual error handling parts and squashed everything into a single commit.
